src/lib_json/json_writer.cpp

```diff
@@ -267,7 +267,7 @@ static String toHex16Bit(unsigned int x) {
   return result;
 }
 
-static String valueToQuotedStringN(const char* value, unsigned length) {
+static String valueToQuotedStringN(const char* value, unsigned length, bool emitUTF8 = false ) {
```

Comment:
It's better to overload it instead of rewriting it, I guess?

Reply:
I want to use the escaping of the tabs, newlines etc. in the first part of the function and then either write the UTF-8 characters or unicode escape them in the second part. Perhaps I should split the function into those separate pieces? What's your view?

Reply:
I would prefer the following solution:

```
static String valueToQuotedStringN(const char* value, unsigned int length) {
    valueToQuotedStringN(value, length, false);
}

static String valueToQuotedStringN(const char* value, unsigned int length, bool emitUTF8) {
...
```

Reply:
I'm glad that you're all considering binary compatibility. Thank you.

Yes, internal "statics" are hidden from the public API, so either way is fine with me.

This change is a nice idea.